main/contracts/contracts.ts

```diff
@@ -61,6 +61,15 @@ export interface IMessage<T = any> {
  * Represents a messagebroker and provides access to the core features which includes publishing/subscribing to messages and RSVP.
  */
 export interface IMessageBroker<T> {
+    /**
+     * A reference to the parent scope if this is not the root node in the tree of scopes. If this is the root, it's undefined.
+     */
+    readonly parent?: IMessageBroker<T>;
```

Comment:
I think the parent may have different generic params. We are assuming all the tree is uniform.  Its fine but something to considerr.

Reply:
If I'm understand right, do you mean supporting something like this?

```typescript
const broker1 = messagebroker<Contract1>();
const broker2 = broker1.createScope<Contract2>('my-scope');
```

in this case, would Contract2 be required to extend Contract 1? Otherwise a message sent to broker1 could not necessarily be passed down to broker2.

Reply:
Yeah so I think this is governed by the propergation logic you plan to implement.  For example say create scope extends the parents channels, then it would mean if I call a channel on the scope it should not be handled locally on the current scope and instead be delgated to the parent